blog/news/urls.py

```diff
@@ -1,13 +1,14 @@
 from django.conf.urls import patterns, include, url
 from django.contrib import admin
-from .views import NewsDetailView
+from .views import NewsDetailView, NewsList
 
 admin.autodiscover()
 
 urlpatterns = patterns('',
     url(r'^admin/', include(admin.site.urls)),
-    url(r'^/?$', 'news.views.news_list', name='news-list'),
+    url(r'^/?$', NewsList.as_view(), name='news-list'),
     url(r'^news/(?P<slug>[\w\-_]+)/$', NewsDetailView.as_view(),
         name='news-detail'),
+    url(r'^(?P<page>[0-9]+)/$', NewsList.as_view(), name='news-list'),
```

Comment:
Bez tego paginacja właśnie nie działa, sprawdzałem. Jeśli użytkownik
nie wywołał którejś kolejnej strony to jest pierwszy widok, jeśli
kliknął na starsze to już przechodzi w ten drugi widok. Myślę, że może
tak być, bo te dwa widoki działają naprzemiennie i w zasadzie są tym
samym widokiem

20.08.2015, geodotka <notifications@github.com> napisał(a):
>>      url(r'^news/(?P<slug>[\w\-_]+)/$', NewsDetailView.as_view(),
>>          name='news-detail'),
>> +    url(r'^(?P<page>[0-9]+)/$', NewsList.as_view(), name='news-list'),
>
> To mi się nie podoba. Póki co nie wiem, co to robi, ale dwa urle nie mogą
> mieć takich samych "name", ponieważ często w templatkach używamy urli po tym
> atrybucie właśnie. Jeśli ma ten url zostać, to "name" trzeba zmienić. Ale
> nie wiem, czy ten url jest konieczny. Chyba ta paginacja powinna działać
> normalnie bez tego, jeśli dobrze mi się wydaje, co to ma robić.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/pawelkowalik/blog/pull/7/files#r37496790